src/libs/elektra/keyset.c

```diff
@@ -243,6 +243,13 @@ KeySet * ksVNew (size_t alloc, va_list va)
 	}
 	ksInit (keyset);
 
+	if (alloc == 0)
```

Comment:
document this feature in the ksNew docu. I think this is a guarantee we can give.

Also add tests for this behavior.

Reply:
There is always a `NULL` at the end AFAIK, so `ks->alloc = 16` means at most 15 keys.

Reply:
> The documentation states that you can allocate a keyset with space for exactly 15 keys, but the allocation size for i.e. ksNew(1, KS_END) is 16. It always defaults to 16 for any alloc < 16. So is this outdated? If so I can rewrite that, too.

Yes, please rewrite it.

Reply:
Yes, but if you call `ksNew(20, KS_END)` at the moment, you'll get a keyset with enough room for 20 keys without reallocation. So I mean the `alloc` you pass to `ksNew` rather than `ks->alloc`.

> Also add tests for this behavior.

How would I test this if it's not visible on the outside?